src/features/animation/hooks/use-video-export.ts

```
@@ -0,0 +1,388 @@
+      trackAnimationEvent("upgrade_prompt_shown", user, {
+        limit_type: "video_exports",
+        trigger: "download_menu",
+      });
+      return {
+        success: false,
+        context: { plan: rpcPlan, current: rpcCurrent, max: rpcMax },
+      };
+    }
+
+    return { success: true };
+  }, [user, usage, supabase, incrementExportCount, decrementExportCount]);
```

Biome (2.1.2)

[error] This variable is used before its declaration.

The variable is declared here:


(lint/correctness/noInvalidUseBeforeDeclaration)